### DISCUSSION TITLE
Nushell startup time

### DISCUSSION THREAD

**Discussion by LamprosPitsillos** [nushell/nushell#12428] [General]
| Command | Mean [ms] | Min [ms] | Max [ms] | Relative |
|:---|---:|---:|---:|---:|
| `dash -c exit` | 1.2 ± 0.1 | 1.0 | 3.8 | 1.00 |
| `zsh --no-rcs -c exit` | 4.4 ± 0.5 | 3.8 | 8.8 | 3.70 ± 0.61 |
| `elvish -norc -c exit` | 4.7 ± 0.6 | 4.0 | 8.9 | 3.99 ± 0.75 |
| `fish --no-config -c exit` | 4.9 ± 0.6 | 4.3 | 9.4 | 4.18 ± 0.74 |
| `bash --norc -c exit` | 6.4 ± 1.7 | 3.2 | 11.0 | 5.44 ± 1.57 |
| `nu --no-config-file -c exit` | 32.2 ± 2.7 | 30.1 | 51.7 | 27.38 ± 4.14 |
| `xonsh --no-rc -c exit` | 585.2 ± 172.2 | 361.4 | 747.0 | 498.01 ± 159.58 |

```
❯ nu -v
0.91.0
```

Is there anything to speedup `nu` startup?

**Comment by fdncred** [comment]
If you're testing pure startup speed, you need to do `nu -n --no-std-lib` because the stdlib is still being loaded and parsed in your test. From what I've seen, nushell's startup time is mainly dependent on the amount of stuff you're sourcing/using in your startup, the speed of your computer and ssd. We know that it's not optimal right now which is one of the reasons why there's working going on in a new parser.

Having said that, I'm not super worried about a 30ms-50ms startup times.

This is what I see on my mac-m1 with `-n --no-std-lib`.
```
> nu -n --no-std-lib
> $nu.startup-time
4ms 684µs 792ns
> exit
> nu -n --no-std-lib
3ms 719µs 375ns
> exit
> nu -n --no-std-lib
3ms 669µs 792ns
```

**Reply by LamprosPitsillos** [reply]
Glad to hear that you are working on improving startup times!
My idea was to use `nu` for scripting, since it looks amazing with it's ability to have command flags and arguments as first class citizens.
So I don't think removing the std-lib is feasible.
I know that 30ms isn't much but ms add up on a script and i would prefer to minimise the delay as much as possible.
Thank you for the response, i will keep a close eye on `nu`
